enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,553 @@
+    crop_suitability['crop_suit'] = (crop_suitability['max_temp'] + crop_suitability['min_temp'] + crop_suitability['temp_range'] + crop_suitability['precip_range'] + crop_suitability['wet_days'] + crop_suitability['max_dry_spells']) / 6
+    crop_suitability = crop_suitability.dropna(dim="year", how="any")
+    return crop_suitability
+
+@APP.callback(
+    Output("timeseries_graph","figure"),
+    Input("loc_marker", "position"),
+    Input("data_choice","value"),
+    Input("submit_params","n_clicks"),
+    State("target_year","value"),
+    State("target_season","value"),
+    State("lower_wet_threshold","value"),
+    State("upper_wet_threshold","value"),
+    State("minimum_temp","value"),
+    State("maximum_temp","value"),
+    State("temp_range","value"),
+    State("season_length","value"),
+    State("min_wet_days","value"),
+    State("wet_day_def","value"),
+    State("number_dry_spells","value"),
+    State("dry_days_in_row","value"),
+    State("dry_spell_rain","value"),
+)
+def timeseries_plot(
+    loc_marker,
+    data_choice,
+    n_clicks,
+    target_year,
+    target_season,
+    lower_wet_threshold,
+    upper_wet_threshold,
+    minimum_temp,
+    maximum_temp,
+    temp_range,
+    season_length,
+    min_wet_days,
+    wet_day_def,
+    number_dry_spells,
+    dry_days_in_row,
+    dry_spell_rain
+):
+    lat1 = loc_marker[0]
+    lng1 = loc_marker[1]
+
+    if data_choice == "suitability_map":
+        data = crop_suitability(
+            rr_mrg, min_wet_days, wet_day_def, tmax_mrg, tmin_mrg, 
+            lower_wet_threshold, upper_wet_threshold, maximum_temp,
+            minimum_temp, temp_range, target_season, dry_spell_rain,
+            dry_days_in_row,number_dry_spells)
+    if data_choice == "precip_map":
+        data = rr_mrg
+    if data_choice == "tmax_map":
+        data = tmax_mrg
+    if data_choice == "tmin_map":
+        data = tmin_mrg
+    
+    try:
+        if data_choice == "precip_map":
+            data_var = pingrid.sel_snap(data.precip, lat1, lng1)
+            isnan = np.isnan(data_var).sum().sum()
+        elif data_choice == "suitability_map":
+            data_var = pingrid.sel_snap(data.crop_suit, lat1, lng1)
+            isnan = np.isnan(data_var).sum().sum()            
+        else:
+            data_var = pingrid.sel_snap(data.temp, lat1, lng1)
+            isnan = np.isnan(data_var).sum().sum()
+        if isnan > 0:
+            error_fig = pingrid.error_fig(error_msg="Data missing at this location")
+            germ_sentence = ""
+            return error_fig, error_fig, germ_sentence
+    except KeyError:
+        error_fig = pingrid.error_fig(error_msg="Grid box out of data domain")
+        germ_sentence = ""
+        return error_fig, error_fig, germ_sentence
+
+    if data_choice == "suitability_map":
+        seasonal_suit = data_var
+        timeseries_plot = pgo.Figure()
+        timeseries_plot.add_trace(
+            pgo.Scatter(
```

Comment:
user Bars for time series with potential missing values in it. See current master onset date for examples